axiom/tests/test_txn.c

```diff
@@ -8068,14 +8084,165 @@ static void test_segment_record_error(void) {
   nr_txn_destroy(&txn);
 }
 
+static nrtxn_t* new_txn_for_record_log_event_test(char* entity_name) {
+  nrapp_t app;
+  nrtxnopt_t opts;
+  nrtxn_t* txn;
+  nr_segment_t* segment;
+
+  /* Setup app state */
+  nr_memset(&app, 0, sizeof(app));
+  app.state = NR_APP_OK;
+  app.entity_name = entity_name;
+
+  /* Setup log feature options */
+  nr_memset(&opts, 0, sizeof(opts));
+  opts.logging_enabled = true;
+  opts.log_forwarding_enabled = true;
+  opts.log_events_max_samples_stored = 10;
```

Comment:
I'm not sure this will have the intended effect as `nr_txn_begin()` will eventually (if not already) use `app->limits.log_events` for the pool size.